backend/backend/views.py

```diff
@@ -67,26 +68,23 @@ def get(self, request, *args, **kwargs):
             BearerTokenSerializer(token).data
             for token in BearerToken.objects.filter(user=request.user).order_by("-created")
         ]
-        try:
-            implicit_token = ImplicitBearerTokenSerializer(ImplicitBearerToken.objects.get(user=request.user)).data
-
-        except ObjectDoesNotExist:
-            implicit_token = None
+        implicit_tokens = [
+            ImplicitBearerTokenSerializer(token).data
+            for token in ImplicitBearerToken.objects.filter(user=request.user).order_by("-created")
+        ]
         return ApiResponse(
             {
                 "tokens": tokens,
-                "implicit_token": implicit_token,
+                "implicit_tokens": implicit_tokens,
             }
         )
 
     def delete(self, request, *args, **kwargs):
-        try:
-            token = BearerToken.objects.get(id=request.GET.get("id"))
-            if request.user == token.user:
-                token.delete()
+        for model in [BearerToken, ImplicitBearerToken]:
+            tokens = model.objects.filter(id=request.GET.get("id"))
+            if len(tokens) == 1 and request.user == tokens[0].user:
```

Comment:
Minor but if we expect only one object, or raise, [get](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#get) does the job

Reply:
My worry is since `len(tokens) == 0` is a common outcome of the above request, I'd be controlling flow with exceptions.

Reply:
better ask for forgiveness than permission ☯️